utilities/ft_affinecoordinates.m

```diff
@@ -0,0 +1,288 @@
+%    it under the terms of the GNU General Public License as published by
+%    the Free Software Foundation, either version 3 of the License, or
+%    (at your option) any later version.
+%
+%    FieldTrip is distributed in the hope that it will be useful,
+%    but WITHOUT ANY WARRANTY; without even the implied warranty of
+%    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%    GNU General Public License for more details.
+%
+%    You should have received a copy of the GNU General Public License
+%    along with FieldTrip. If not, see <http://www.gnu.org/licenses/>.
+%
+% $Id$
+
+% these are the 48 generic axis orientation triplets, these specify the axes but no origin
+%   a = anterior
+%   p = posterior
+%   l = left
+%   r = right
+%   s = superior
+%   i = inferior
+
+generic = {
+  'als'; 'ali'; 'ars'; 'ari';...
+  'pls'; 'pli'; 'prs'; 'pri';...
+  'las'; 'lai'; 'ras'; 'rai';...
+  'lps'; 'lpi'; 'rps'; 'rpi';...
+  'asl'; 'ail'; 'asr'; 'air';...
+  'psl'; 'pil'; 'psr'; 'pir';...
+  'sal'; 'ial'; 'sar'; 'iar';...
+  'spl'; 'ipl'; 'spr'; 'ipr';...
+  'sla'; 'ila'; 'sra'; 'ira';...
+  'slp'; 'ilp'; 'srp'; 'irp';...
+  'lsa'; 'lia'; 'rsa'; 'ria';...
+  'lsp'; 'lip'; 'rsp'; 'rip'}';
+
+% the specific ones specify the axes and also the origin
+specific = {'ctf', 'bti', 'fourd', 'yokogawa', 'eeglab', 'neuromag', 'itab', 'acpc', 'spm', 'mni', 'fsaverage', 'tal', 'scanras', 'scanlps', 'dicom'};
+
+%--------------------------------------------------------------------------
+% These approximate alignments are based on transformation matrices that were
+% determined by clicking on the CTF fiducial locations in the canonical T1 template
+% MRI.
+%
+% All of the transformation matrices here are expressed in millimeter.
+
+% this is based on the ear canals, see ALIGN_CTF2ACPC
+acpc2ctf = [
+   0.0000  0.9987  0.0517  34.7467
+  -1.0000  0.0000  0.0000   0.0000
+   0.0000 -0.0517  0.9987  52.2749
+   0.0000  0.0000  0.0000   1.0000
+  ];
+
+% this is based on the ear canals, see ALIGN_NEUROMAG2ACPC
+acpc2neuromag = [
+  1.0000  0.0000  0.0000   0.0000
+  0.0000  0.9987  0.0517  34.7467
+  0.0000 -0.0517  0.9987  52.2749
+  0.0000  0.0000  0.0000   1.0000
+  ];
+
+% see http://freesurfer.net/fswiki/CoordinateSystems
+fsaverage2mni = [
+   0.9975   -0.0073    0.0176   -0.0429
+   0.0146    1.0009   -0.0024    1.5496
+  -0.0130   -0.0093    0.9971    1.1840
+   0.0000    0.0000    0.0000    1.0000
+  ];
+
+% this is a 90 degree rotation around the z-axis
+ctf2neuromag = [
+  0.0000   -1.0000    0.0000    0.0000
+  1.0000    0.0000    0.0000    0.0000
+  0.0000    0.0000    1.0000    0.0000
+  0.0000    0.0000    0.0000    1.0000
+  ];
+
+% these are all combinations of 90 degree rotations and/or flips along one of the axes
+for i=1:length(generic)
+  for j=1:length(generic)
+    xxx = generic{i};
+    yyy = generic{j};
+    eval(sprintf('%s2%s = transform_generic(''%s'', ''%s'');', yyy, xxx, xxx, yyy));
```

Comment:
I don't fully understand the order: yyy,xxx,xxx,yyy (i.e. the swap for the second pair). This suggests that what is known in transform_generic as 'from' should actually be 'to', or vice versa. Or am I missing something? 

Note that the code behaves properly, so it's just probably some confusion on my end

Reply:
yes, you seem to be right about the to/from swap in `transform_generic`. Let me fix it.